File: 202102653/1654.cpp
#include<iostream>

// Declare variable
unsigned int numberOfHavingWire, numberOfNeedWire, ans;
unsigned int wireArr[10001];
int  max = 0;
long long mid, high, low;

int main()
{
	std::cin >> numberOfHavingWire >> numberOfNeedWire;

    if(numberOfHavingWire > numberOfNeedWire) {
        std::cout << "Error : Invalid Input" << std::endl;
        return 1;
    }

	for (unsigned int i = 0; i < numberOfHavingWire; i++)
	{
		std::cin >> wireArr[i];
		if (max < wireArr[i]) max = wireArr[i];
	}

	low = 1;
	high = max;
	ans = 0;

	while (low <= high)
	{
		mid = (low + high) / 2;
		int cnt = 0;
		for (unsigned int i = 0; i < numberOfHavingWire; i++)
			cnt += wireArr[i] / mid;

		if (cnt >= numberOfNeedWire) {
			low = mid + 1;
			if (ans < mid) ans = mid;
		} else {
			high = mid - 1;
		}
	}
	
    if(ans == 0) {
        std::cout << "Error: Cannot create as many wires as desired" << std::endl;
            return 1;
    }

	std::cout << ans << std::endl;

    return 0;
}

File: 201902658/1654.py
a, b = list(map(int, input().split()))

input_list = []

for i in range(a):
  length = int(input())
  input_list.append(length)

start = 1;
end = max(input_list)

while (start <= end):
    mid = (start + end) // 2
    cnt = 0
    for i in range(a):
        cnt += input_list[i] // mid
    if cnt >= b:
        start = mid + 1
    else:
        end = mid - 1
print(end)